crates/server/src/api/routes/quests/create_quest.rs

```diff
@@ -17,21 +16,59 @@ pub struct CreateQuestResponse {
 }
 
 #[derive(Deserialize, Serialize, ToSchema, Debug)]
+#[serde(rename_all = "camelCase")]
 pub struct QuestReward {
     pub hook: QuestRewardHook,
     pub items: Vec<QuestRewardItem>
 }
 
-#[derive(Deserialize, Serialize, ToSchema, Deref, Debug)]
+#[derive(Deserialize, Serialize, ToSchema, Debug)]
+#[serde(rename_all = "camelCase")]
 pub struct CreateQuestRequest {
     pub name: String,
     pub description: String,
-    #[deref]
     pub definition: QuestDefinition,
     pub image_url: String,
     pub reward: Option<QuestReward>,
 }
 
+impl CreateQuestRequest {
+    pub fn is_valid(&self) -> Result<(), QuestError> {
+        if self.name.trim().len() < 5 {
+            return Err(QuestError::QuestValidation("Name should be longer".to_string()))
+        }
+
+        if self.description.trim().len() < 10 {
```

Comment:
Where did the magic numbers (5 and 10) come from?

Reply:
nowhere, it's just to make the name and description useful. I'm replacing 10 with 5 for desc